compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/create_aie_workgroup.mlir

```diff
@@ -0,0 +1,383 @@
+// CHECK-SAME:    [0, 0, 0, 0] [1, 1, 8, 16] [128, 128, 16, 1]
+// CHECK-SAME:    [0, 0, 0, 0] [1, 1, 8, 16] [128, 16, 16, 1]
+// CHECK-SAME:    (!amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 2>>, !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>)
+func.func @circular_dma_cpy_nd(%arg0: !amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 2>>, %arg1: !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>) {
+  %0 = amdaie.circular_dma_cpy_nd(%arg0[0, 0, 0, 0] [1, 1, 8, 16] [128, 128, 16, 1], %arg1[0, 0, 0, 0] [1, 1, 8, 16] [128, 16, 16, 1]) : (!amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 2>>, !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>)
+  return
+}
+
+// -----
+
+// CHECK-LABEL: @core
+// CHECK:       amdaie.workgroup
+// CHECK:         %[[TILE_0:.+]] = amdaie.tile
+// CHECK:         %[[TILE_1:.+]] = amdaie.tile
+// CHECK:         %{{.+}} = amdaie.core(%[[TILE_0]])
+// CHECK:         %{{.+}} = amdaie.core(%[[TILE_1]])
+// CHECK:         amdaie.controlcode
+func.func @core() {
+  %c0 = arith.constant 0 : index
+  %c1 = arith.constant 1 : index
+  %tile_0_0 = amdaie.tile(%c0, %c0)
+  %tile_0_1 = amdaie.tile(%c0, %c1)
+  %core_0_0 = amdaie.core(%tile_0_0) {
+    amdaie.end
+  }
+  %core_0_1 = amdaie.core(%tile_0_1) {
+    amdaie.end
+  }
+  return
+}
+
+// -----
+
+// CHECK-LABEL: @dma_cpy_nd
+// CHECK:       amdaie.workgroup
+// CHECK-DAG:     %[[FROMMEMREF0:.+]] = amdaie.logicalobjectfifo.from_memref
+// CHECK-SAME:    memref<1x1x8x16xi32, 2> -> !amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 2>>
+// CHECK-DAG:     %[[FROMMEMREF1:.+]] = amdaie.logicalobjectfifo.from_memref
+// CHECK-SAME:    memref<8x16xi32, 1> -> !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>
+// CHECK:         %[[DMA:.+]] = amdaie.circular_dma_cpy_nd
+// CHECK-SAME:    %[[FROMMEMREF0]][] [] []
+// CHECK-SAME:    %[[FROMMEMREF1]][] [] []
+// CHECK-SAME:    (!amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 2>>, !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>)
+// CHECK:         amdaie.controlcode
+// CHECK:           %[[IPU_DMA:.+]] = amdaie.npu.dma_cpy_nd %[[DMA]]
+// CHECK-SAME:      [] [] []
+// CHECK-SAME:      [0, 0, 0, 0] [1, 1, 8, 16] [128, 16, 16, 1]
+// CHECK:           amdaie.npu.dma_wait(%[[IPU_DMA]], S2MM)
+func.func @dma_cpy_nd(%arg0: memref<1x1x8x16xi32, 2>, %arg1: memref<8x16xi32, 1>) {
+  %0 = amdaie.logicalobjectfifo.from_memref %arg0, {} : memref<1x1x8x16xi32, 2> -> !amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 2>>
+  %1 = amdaie.logicalobjectfifo.from_memref %arg1, {} : memref<8x16xi32, 1> -> !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>
+  %2 = amdaie.dma_cpy_nd(%0[] [] [], %1[0, 0, 0, 0] [1, 1, 8, 16] [128, 16, 16, 1]) : (!amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 2>>, !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>)
+  return
+}
+
+// -----
+
+// CHECK-LABEL: @for
+// CHECK:       amdaie.workgroup
+// CHECK-DAG:     %[[C0:.+]] = arith.constant 0 : index
+// CHECK-DAG:     %[[C1:.+]] = arith.constant 1 : index
+// CHECK-DAG:     %[[C8:.+]] = arith.constant 8 : index
```

Comment:
> those cases where we are directly using %c1, %c128, etc.

What do you exactly mean with this? Do you mean add a CHECK where this is used in some operation?

Reply:
Yes, this lit test check statements are absolutely correct.

Like how you've captured the constant value using `CHECK-DAG` and used them where they appear is how we should be doing for other lit tests too - instead of directly matching the SSA appearances.

Reply:
I see. Will check the other lit tests.

Reply:
Updated the ones I added in `roundtrip.mlir`. I will create a follow-up for other ones that have already been merged.